### ISSUE TITLE
Publishing to Maven Central is broken

### ISSUE THREAD

**Issue by krzema12** [typesafegithub/github-workflows-kt#1501]
So far, manifests itself only in snapshot publishing, but regular releasing may be affected as well.

```
> Task :shared-internal:publishMavenJavaPublicationToSonatypeRepository FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':shared-internal:publishMavenJavaPublicationToSonatypeRepository'.
> Failed to publish publication 'mavenJava' to repository 'sonatype'
   > Could not PUT 'https://s01.oss.sonatype.org/content/repositories/snapshots/io/github/typesafegithub/shared-internal/2.1.1-SNAPSHOT/shared-internal-2.1.1-20240614.1[45](https://github.com/typesafegithub/github-workflows-kt/actions/runs/9518067962/job/26238109523#step:6:46)042-13.jar'. Received status code 401 from server: Content access is protected by token
```

Example job: https://github.com/typesafegithub/github-workflows-kt/actions/runs/9518067962/job/26238109523

**Comment by krzema12** [comment]
https://help.sonatype.com/en/user-tokens.html

**Comment by krzema12** [comment]
Despite using an access token, it still fails. Maybe some cache still keeps the old creds.

**Comment by krzema12** [comment]
I wrote to Central Support (central-support@sonatype.com).

**Comment by ZebNoid** [comment]
Sonatype ditched user password authentication (for security reasons)

```
Dear Maven Central publisher,

We are making changes to the OSSRH authentication backend. For most users this should be a transparent process, and you should be able to continue to use your existing username and password to connect the Nexus UI. In case you need to update your password, please follow our documentation.

To configure a publisher’s plugin authentication you would need to update your plugin settings to use a user token instead of the Nexus UI username and password login.
```

Unhelpful instructions [401 Content access is protected by token when accessing repositories](https://support.sonatype.com/hc/en-us/articles/360049469534-401-Content-access-is-protected-by-token-when-accessing-repositories) 

Helpful ["401 Content access is protected by token" authentication failure while performing maven release](https://community.sonatype.com/t/401-content-access-is-protected-by-token-authentication-failure-while-performing-maven-release/12741/4)

In short

1. Go to [https://s01.oss.sonatype.org/ 4](https://s01.oss.sonatype.org/)
2. Go to profile
3. Change the pulldown from “Summary” to “User Token”
4. Click on “Access User Token”

```gradle
maven {
    credentials {
        username = "your_token"
        password = "your_token_password"
    }
}
```

**Comment by krzema12** [comment]
@ZebNoid thanks! I was doing the same steps you suggested, but with one difference: instead of https://s01.oss.sonatype.org/ I was using https://oss.sonatype.org/. Looks like they use separate token spaces.

Successful publish: https://github.com/typesafegithub/github-workflows-kt/actions/runs/9561573715/job/26431480899
